### ISSUE TITLE
Potential slowdown when fetching a list of transactions

### ISSUE THREAD

**Issue by DanReyLop** [Automattic/woocommerce-payments#132]
See this conversation for more context: https://github.com/Automattic/woocommerce-payments/pull/127#discussion_r300384153

TL;DR: When fetching a list of transactions from wp.com, **for each transaction** we're making 2 database queries:
- One to fetch the order id associated with that Stripe transaction.
- One to fetch the order from that order ID

For long lists of transactions, that can add up to a significant amount of time on low-end hosts.

Possible mitigations are:
- Combine all the queries to fetch the order IDs into a single query that uses `meta_value IN ( charge_id_1, charge_id_2, etc )`.
- Not fetch the Order object at all, it's not needed to get the "order number" (which is just the order ID) or a link to the edit order screen (which just depends on the order ID). Note that this may break interoperability with extensions that rely on filtering those values, like "Sequential Order Numbers". We could potentially only fetch the Order object if we detect those filters are being used.
